### PR TITLE
Adding documentation on monotonicity.

### PR DESCRIPTION
**Pull request by cmungall**
Aligning monotonicity tests with jsonschema behavior

See also https://github.com/linkml/linkml/issues/1483

### CODE REVIEW

File: docs/schemas/advanced.md [hunk 1]
```diff
@@ -53,13 +93,14 @@ enums:
       UNDEAD:
 ```
 
-Note that these constructs may ignored by some generators in the linkml 1.1 series.
+Note that the range of `vital_status` is declared as `Any`, which is further constrained by the `any_of` expression.
+
+Currently, it is important to have this base range declaration, because LinkML constraint semantics are
```
Comment (dalito) [inline docs/schemas/advanced.md:98]: What exactly is meant with "base range declaration", is it "range:any"?

File: docs/schemas/advanced.md [hunk 2]
```diff
@@ -53,13 +93,14 @@ enums:
...
+monotonic. If this were not stated, then the `default_range` of string would be applied.
```
Comment (dalito) [inline docs/schemas/advanced.md:99]: "monotonic" may be unclear for non-experts. Having an explanation would be helpful. I did not know about the non-math meaning of "monotonic" before you used it in the discussions some time ago.

**Review by dalito — COMMENTED** [review]
Just some hints were I thing it the explanation could be even clearer.

**Review by sierra-moxon — APPROVED** [review]

File: docs/schemas/advanced.md (hunk 2 above)
Comment (cmungall) [reply docs/schemas/advanced.md]: Thx! Linked back to slots docs for this, and added a bit of qualification

**Pull request merged**